### ISSUE TITLE
How learning rate scheduler works with Custom training loop using tf.GradientTape()

### ISSUE THREAD

**Issue by kamalkraj** [tensorflow/models#7687]
Do I need to manually invoke the scheduler with steps information?
How the scheduler gets the steps information?
Is there any working example available?

Example Problem : 
```python
# Instantiate an optimizer.
learning_rate_fn = tf.keras.optimizers.schedules.PolynomialDecay(
      initial_learning_rate=init_lr,
      decay_steps=num_train_steps,
      end_learning_rate=0.0)
optimizer = tf.keras.optimizers.SGD(learning_rate=learning_rate_fn)
loss_fn = keras.losses.SparseCategoricalCrossentropy(from_logits=True)

# Iterate over the batches of a dataset.
for x_batch_train, y_batch_train in train_dataset:
  with tf.GradientTape() as tape:
    logits = layer(x_batch_train)  # Logits for this minibatch
    # Loss value for this minibatch
    loss_value = loss_fn(y_batch_train, logits)
    # Add extra losses created during this forward pass:
    loss_value += sum(model.losses)

  grads = tape.gradient(loss_value, model.trainable_weights)
  optimizer.apply_gradients(zip(grads, model.trainable_weights))
```

**Comment by saberkun** [comment]
optimizer has a tensor: iterations which is the optimizer global_step.
learning_rate_fn access it to update learning rate I think.
To sanity check, you can print learning rate as: 
current_step = xxx or optimizer.iterations.numpy()
tf.summary.scalar(
          "learning_rate", optimizer._decayed_lr(var_dtype=tf.float32), 
          step=current_step)

**Comment by innat** [comment]
FYI, 

I've encountered this [answer](https://stackoverflow.com/a/61553047/9215780) from SO. In summary:


```python
train_step_count = 0
summary_writer = tf.summary.create_file_writer('logs/')
def train_step(images, labels):
  train_step_count += 1
  with tf.GradientTape() as tape:
    predictions = model(images)
    loss = loss_object(labels, predictions)
  gradients = tape.gradient(loss, model.trainable_variables)
  optimizer.apply_gradients(zip(gradients, model.trainable_variables))

  # optimizer._decayed_lr(tf.float32) is the current Learning Rate.
  # You can save it to TensorBoard like so:
  with summary_writer.as_default():
    tf.summary.scalar('learning_rate',
                      optimizer._decayed_lr(tf.float32), # <---------- optimizer._decayed_lr()
                      step=train_step_count)
```
